pandas/tests/indexes/categorical/test_indexing.py

```diff
@@ -298,6 +298,18 @@ def test_get_indexer_same_categories_different_order(self):
         expected = np.array([1, 1], dtype="intp")
         tm.assert_numpy_array_equal(result, expected)
 
+    def test_get_indexer_nans_in_index_and_target(self):
+        # GH 45361
+        ci = CategoricalIndex([1, 2, np.nan, 3])
+        other1 = [2, 3, 4, np.nan]
+        other2 = [1, 4, 2, 3]
```

Comment:
can you put the  all of the foo2 lines below all of the foo1 lines

Reply:
Done